client/http/client.go

```diff
@@ -29,16 +30,29 @@ type HttpClientConfig struct {
 	RestClient  *resty.Client
 }
 
+func getJWTToken(httpClient *HttpClient) (*resty.Response, error) {
+	req := httpClient.client.R().SetBasicAuth(httpClient.ApiKey, httpClient.ApiSecret)
+	response, err := req.SetQueryParams(map[string]string{"encoded": "true"}).Get("auth/token")
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
 func NewHttpClient(config HttpClientConfig) (*HttpClient, error) {
-	return &HttpClient{
+	httpClient := &HttpClient{
 		ApiKey:    config.ApiKey,
 		ApiSecret: config.ApiSecret,
 		client:    config.RestClient.SetHostURL(config.ApiEndpoint).SetHeader("User-Agent", config.UserAgent),
-	}, nil
+	}
+	response, err := getJWTToken(httpClient)
+
+	httpClient.jwtToken = string(response.Body())
+	return httpClient, err
```

Comment:
You can't return both an `err` and a valid response, it goes against the format...

```suggestion
	return httpClient, nil

Reply:
[fix error return](https://github.com/env0/terraform-provider-env0/pull/495/commits/9fe7e97c5fbc0a0dc360d6df31627a72d0b9d6fd)